docs/docs/docs/developer-resources/testing.md

```diff
@@ -17,42 +17,119 @@ All our workflows use Linux based commands, therefore if you are a developer who
+You can run these scripts yourself to ensure that your pull request doesn't fail due to linting and formatting errors:
 
+```bash
+npm run lint:check
+npm run format:check
 ```
 
-### Debugging tests
+## Vitest Testing
+
+This project uses [Vitest](https://vitest.dev/) as the testing framework.
+
+### Running Tests
+
+- **Run all tests:**
+  ```bash
+  npm run test
+  ```
+
+- **Run a single test file:**
+  ```bash
+  npm run test /path/to/test/file
+  ```
+
+- **Watch tests for changes:**
+  ```bash
+  npm run test:watch
+  ```
+  This opens the Vitest UI for interactive testing.
+
+### Test Coverage
+
+- **View coverage for all test files:**
+  ```bash
+  npm run test:coverage
+  ```
+
+- **View coverage for a single test file:**
+  ```bash
+  npm run test:coverage /path/to/test/file
+  ```
+
+- **Generate HTML coverage report:**
+  ```bash
+  npm run test:coverage
+  genhtml coverage/lcov.info -o coverage
+  ```
+  The `genhtml` command is part of the Linux `lcov` package. Similar packages are available for Windows and MacOS.
 
-You can see the output of failing tests in broswer by running `jest-preview` package before running your tests
+### Test Sharding
+
+For improved performance, especially in CI/CD environments, you can run tests in parallel using sharding:
 
 ```bash
-npm run jest-preview
-npm run test
+npm run test:shard
+```
 
+To run a specific shard, use environment variables:
+
+```bash
+SHARD_INDEX=1 SHARD_COUNT=4 npm run test:shard
 ```
 
-You don't need to re-run the `npm run jest-preview` command each time, simply run the `npm run test` command if the Jest Preview server is already running in the background, it'll automatically detect any failing tests and show the preview at `http://localhost:3336` as shown in this screenshot -
+This divides the test suite into 4 parts and runs the 1st part. This is particularly useful for:
+- Running tests in parallel across multiple CI jobs
+- Faster test execution on multi-core systems
+- Reducing overall test suite runtime
 
-![Debugging Test Demo](../../../static/img/markdown/installation/jest-preview.webp)
+You can also run sharded tests with coverage:
+
+```bash
+npm run test:shard:coverage
+```
 
-### Linting code files
+### Code Coverage Standards
 
-You can lint your code files using this command:
+- The current code coverage of the repository: [![codecov](https://codecov.io/gh/PalisadoesFoundation/talawa-admin/branch/develop/graph/badge.svg?token=II0R0RREES)](https://codecov.io/gh/PalisadoesFoundation/talawa-admin)
+- The currently acceptable coverage rate can be found in the [GitHub Pull Request workflow file](.github/workflows/pull-requests.yml). Search for the value below the line containing `min_coverage`.
+
+## Cypress End-to-End Testing
+
+Cypress is used for end-to-end testing to ensure the application works correctly from a user's perspective.
+To read more about Cypress testing, please refer to the [Cypress Guide](../../../../cypress/README.md).
```

Comment:
This should read:

```
To read more about Cypress testing, please refer to the [Our end to end testing with Cypress guide](e2e-testing.md).
```

Docusaurus can't reference links above `docs/` even though your editor can. This is why the website deployment test is failing.

Please look at this file and run Docusaurus locally to validate the links. 

```
docs/README.md
```

On startup there should be no errors like this:

<img width="2455" height="513" alt="Image" src="https://github.com/user-attachments/assets/326cc013-82e9-4d09-a44b-a8667a8460a5" />